### DISCUSSION TITLE
Update an existing event's goal props / Send on unload or visibilitychange

### DISCUSSION THREAD

**Discussion by abett** [plausible/analytics#1494] [Anything Else]
Hi!

I'm wondering whether there is a possibility for me to update the properties of custom events.  
Imagine the events "Video Playback" or "Page Scroll Depth" with the prop "percentage" being updated occasionally.

I've currently implemented the code as suggested for scroll depth [here](https://github.com/plausible/analytics/discussions/121#discussioncomment-1219503) and I'm not 100% on board with tracking every 10% step as a separate event. That just floods the dashboard and it's not really nice to analyze, either.

(Alternatively, it would solve my problem if I was able to reliably send tracking events on `unload` or `visibilitychange`. Is that possible?)

**Comment by ukutaht** [comment]
To provide something like this, we need to solve 2 separate problems:
1. Sending events reliably during/after `unload`
2. Being able to update events in the database post-collection (analytical data is not so easy to mutate)

We will start tackling both of these this winter. @RobertJoonas will be working on a solution to both of these problems. For reliable delivery we will be evaluating the `Navigator.sendBeacon` API. At first, it will only be used to capture time on page and visit length more accurately than current methods. This implementation can then be extended to allow custom data to be sent during `unload`.

**Reply by abett** [reply]
thanks for the insight! wishing you the smoothest of projects, @RobertJoonas 🔥

**Reply by abett** [reply]
(any github issue I can subscribe to to stay updated on this topic?)

**Reply by ukutaht** [reply]
Thanks @abett. We'll post updates here: https://github.com/plausible/analytics/discussions/190
